github/client.go

```diff
@@ -544,14 +551,20 @@ func (client *Client) FindOrCreateToken(user, password, twoFactorCode string) (t
 	return
 }
 
-func (client *Client) api() (c *octokit.Client, err error) {
+func (client *Client) ensureAccessToken() (err error) {
 	if client.Host.AccessToken == "" {
-		host, e := CurrentConfig().PromptForHost(client.Host.Host)
-		if e != nil {
-			err = e
-			return
+		host, err := CurrentConfig().PromptForHost(client.Host.Host)
```

Comment:
`:=` creates a local variable `err` which is not the same as the one to return `(err, error)`.  Currently `(err, error)` is always `nil`. We could directly return the err if it's not nil:

```go
host, err := CurrentConfig().PromptForHost(client.Host.Host)
if err == nil {
  client.Host = host
} else {
  return err
}

return nil
```

Or like before, use a different local variable name and assign it to `(err error)`:

```go
host, e := CurrentConfig().PromptForHost(client.Host.Host)
if e != nil {
  err = e
  return
} 

client.Host = host
...
```